### PR TITLE
Bug fix/dont create dangling branches

### PR DESCRIPTION
**Pull request by goedderz**
_The core fix was merged with #10267 already._

### Scope & Purpose

When optimizing the Ast, a ternary expression with a constant condition is replaced by a reference to the particular subquery. When instantiating the plan, the ternary expression was omitted (which is fine), and the subquery changed to directly write the result variable of the ternary expression. However, the plan building would not continue at the current node, but at the subquery node, which leaves a dangling branch behind and prevents the other subquery from being executed.

- [X] Bug-Fix for *devel-branch*
- [X] Bug-Fix for *3.6*: https://github.com/arangodb/arangodb/pull/10679
- [X] Bug-Fix for *3.5*: https://github.com/arangodb/arangodb/pull/10531
- [X] Bug-Fix for *3.4*: https://github.com/arangodb/arangodb/pull/10586
- [X] The behavior in this PR can be (and was) *manually tested*
- [X] The behavior change can be verified via automatic tests

### Testing & Verification

This PR adds tests that were used to verify all changes:

- [X] Added **Regression Tests** (Only for bug-fixes) 

### Documentation

- [X] Added a *Changelog Entry*

### CODE REVIEW

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6839/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6848/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6984/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7372/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7404/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7419/

**Comment by goedderz** [comment]
Last Jenkins is blue:
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7443/

File: arangod/Aql/Optimizer.cpp [hunk 1]
```diff
@@ -70,13 +70,117 @@ void Optimizer::addPlanAndRerun(std::unique_ptr<ExecutionPlan> plan,
   addPlanInternal(std::move(plan), rule, wasModified, it);
 }
 
+
+#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
+
+// Check the plan for inconsistencies, like more than one parent or dependency,
+// or mismatching parents and dependencies in adjacent nodes.
+class CheckPlan : public WalkerWorker<ExecutionNode> {
```
Comment (jsteemann) [inline arangod/Aql/Optimizer.cpp]: Let's rename this `PlanChecker`

Comment (goedderz) [reply arangod/Aql/Optimizer.cpp]: Done.

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7461/

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7644/

**Comment by goedderz** [comment]
Jenkins is blue.

**Pull request merged**